pyomo/contrib/mindtpy/util.py

```diff
@@ -844,3 +849,65 @@ def set_up_logger(config):
     ch.setFormatter(formatter)
     # add the handlers to logger
     config.logger.addHandler(ch)
+
+
+def get_dual_integral(solve_data):
+    dual_integral = 0
+    dual_bound_progress = solve_data.dual_bound_progress.copy()
+    # Initial dual bound is set to inf or -inf. To calculate dual integral, we set
+    # initial_dual_bound to 10% greater or smaller than the first_found_dual_bound.
+    # TODO: check if the calculation of initial_dual_bound needs to be modified.
+    for dual_bound in dual_bound_progress:
+        if dual_bound != dual_bound_progress[0]:
+            break
+    for i in range(len(dual_bound_progress)):
+        if dual_bound_progress[i] == solve_data.dual_bound_progress[0]:
+            if solve_data.objective_sense == minimize:
+                if dual_bound > 0:
+                    dual_bound_progress[i] = dual_bound * 0.9
```

Comment:
`dual_bound_progress[i] = dual_bound * (1 + 0.1**solve_data.objective_sense)` avoids the conditional. We can consider adding that 0.1 as a parameter